auth.go

```diff
@@ -10,17 +10,19 @@ import (
 
 //AddBasicAuth to a request by reading secrets
 func AddBasicAuth(req *http.Request) error {
-	reader := auth.ReadBasicAuthFromDisk{}
+	if os.Getenv("basic_auth") == "true" {
```

Comment:
This code is not very efficient since it may read the file on every result that is posted back etc. We'd be better off reading the credentials once at the top of the main.go and then passing them in to this method - if they are nil, we do nothing, if set we apply them.

What do you think?

Reply:
I think we can load this at the start of the worker and use it.